halpybot/commands/caseutils.py

```diff
@@ -7,21 +7,60 @@
 Licensed under the GNU General Public License
 See license.md
 """
-from typing import List, Union
+import functools
+import re
+from typing import List
 from pendulum import now
+from halpybot import config
 from ..packages.command import Commands, get_help_text
-from ..packages.models import Context, User, NoUserFound, Case
+from ..packages.edsm import (
+    sys_cleaner,
+    checklandmarks,
+    NoResultsEDSM,
+    EDSMConnectionError,
+)
+from ..packages.models import (
+    Context,
+    User,
+    NoUserFound,
+    Case,
+    Status,
+    CaseType,
+    Platform,
+    KFCoords,
+    KFType,
+)
+from ..packages.checks import Require, Drilled
+from ..packages.case import get_case, update_single_elem_case_prep
 
 
-async def get_case(ctx: Context, case_arg: str) -> Case:
-    """Fetch a case from the Board"""
-    try:
-        caseref: Union[str, int] = int(case_arg)
-    except ValueError:
-        caseref = case_arg
-    return ctx.bot.board.return_rescue(caseref)
+# Decorators
+class CaseUtils:
+    """Utilities for Wrapping EDSM Commands"""
+
+    @staticmethod
+    def gather_case(len_args_expected: int):
+        """Process gathering Case details for a command expecting a number of args"""
+
+        def decorator(function):
+            @functools.wraps(function)
+            async def guarded(ctx, args: List[str]):
+                if len(args) < len_args_expected:
+                    return await ctx.reply(
+                        get_help_text(ctx.bot.commandsfile, ctx.command)
+                    )
+                try:
+                    case: Case = await get_case(ctx, args[0])
+                except KeyError:
+                    return await ctx.reply(f"No case found for {args[0]!r}.")
+                return await function(ctx, args, case)
```

Comment:
recommendation: pass `case` by kwarg to the underlying.

Reply:
Fixed in ac79255 (#338)